server/api/dataset.py

```diff
@@ -425,19 +425,14 @@ def _checkFileFormat(self, file, formats):
     )
     @access.cookie
     @access.public(scope=TokenScope.DATA_READ)
-    @loadmodel(model='file', map={'metadataFileId': 'metadataFile'}, force=True, exc=False)
+    @loadmodel(model='file', map={'metadataFileId': 'metadataFile'}, level=AccessType.READ,
```

Comment:
I think this should be `WRITE` access to the file (which is the same as `WRITE` access to the dataset) in order to download the raw CSV file. Normal users of the dataset (with `READ` access) have no need to access the CSV itself.

Reply:
Also, can you add a comment, noting the fact that this is really `WRITE` access to the dataset? Since we don't use the permission checking of `attachedTo` files much, I don't want to forget where this permission comes from. Also, add a comment to other places in this file where do you do a similar thing.